### PR TITLE
[IMP] kyocera_free_shipping_notice: Alert when amount less than 99

### PR DESCRIPTION
**Pull request by yall-odoo**
### Description

Added the necessary code to show an alert if the amount will allow for free shipping (orders > 99). If not, the website will show an alert on the cart page where we are prompted to buy an additional product to reach 99 and get free shipping.

Missing: If only the quantity of a product is updated, the remaining amount for free shipping is not updated until a hard refresh.

Link to task: [2944841](https://www.odoo.com/web#model=project.task&id=2944841)

### All Submissions:

* [x] My commit respects the [Odoo commit guideline](https://www.odoo.com/documentation/15.0/developer/misc/other/guidelines.html#git)
* [x] My commit message respects the [commit template](https://github.com/odoo-ps/psbe-process/wiki/Commits-message-guidelines#template)
* [ ] I have used pre-commit
* [x] The PR contains **only** my modification and **no other external** commit

### Sh/Runbot:

* [ ] The commits pass test and the branch is green
* [ ] Unit tests have been implemented / standard ones rewritten
* [ ] The Staging is ISO-Prod and will contain only this dev

### Upgrade:

* [ ] The data affected (*if any*) by the changes has been migrated

### CODE REVIEW

File: kyocera_free_shipping_notice/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+============================
+This module contains the elements necessary for the display of a popup displaying the remaining amount
+to get free shipping if the user is spending less than 99$ on their current order.
+Author: yall
+Task: 2944841
+  ''',
+  'author':'Odoo PS',
+  'category':'ecommerce',
+  'version':'15.0.0.1',
+  'depends':['website_sale'],
```
Comment (tdetry) [inline kyocera_free_shipping_notice/__manifest__.py:15]: i like when it is formatted the same way as the 'data'

File: kyocera_free_shipping_notice/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    remaining_for_free_shipping = fields.Float('Remaining amount for free shipping')
+
+    @api.depends('order_line')
+    def _compute_website_order_line(self):
+        super()._compute_website_order_line()
+        for order in self:
+            if order.amount_untaxed < FREE_SHIP_AMT:
+                order.remaining_for_free_shipping = FREE_SHIP_AMT - order.amount_untaxed
+            else:
+                order.remaining_for_free_shipping = 0
```
Comment (tdetry) [inline kyocera_free_shipping_notice/models/sale_order.py:18]: NO ! please see slide 8 (last point)  https://docs.google.com/presentation/d/1p-Z18_vM_groCV2s2Ajv5NCBFkNQWfXKPNThRra0PrM/edit#slide=id.g13a1f6d23c5_0_49

File: kyocera_free_shipping_notice/views/templates_inherit.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <template id="cart_inherit" inherit_id="website_sale.cart" name="Shopping Cart Inherit">
+        <t t-call="website_sale.cart_lines" position="after">
+            <div t-if="website_sale_order.remaining_for_free_shipping > 0" class="alert alert-info">
+                Spend <t t-out="website_sale_order.remaining_for_free_shipping"/> dollars more to get free shipping!
+            </div>
+            <div t-else="website_sale_order.is_free_shipping" class="alert alert-info">
```
Comment (tdetry) [inline kyocera_free_shipping_notice/views/templates_inherit.xml:8]: where is is_free_shipping ?

**Review by tdetry — CHANGES_REQUESTED** [review]
improvements needed!

File: kyocera_free_shipping_notice/models/sale_order.py (hunk 2 above)
Comment (yall-odoo) [reply kyocera_free_shipping_notice/models/sale_order.py:18]: OMG sorry about that

File: kyocera_free_shipping_notice/views/templates_inherit.xml (hunk 3 above)
Comment (yall-odoo) [reply kyocera_free_shipping_notice/views/templates_inherit.xml:8]: I missed it while reviewing my code and forgot to erase it

File: kyocera_free_shipping_notice/models/sale_order.py [hunk 4]
```diff
@@ -6,11 +6,10 @@
...
     _inherit = 'sale.order'
 
-    remaining_for_free_shipping = fields.Float('Remaining amount for free shipping')
+    remaining_for_free_shipping = fields.Float('Remaining amount for free shipping', compute='_compute_remaining_for_free_shipping')
 
     @api.depends('order_line')
-    def _compute_website_order_line(self):
-        super()._compute_website_order_line()
+    def _compute_remaining_for_free_shipping(self):
         for order in self:
```
Comment (tdetry) [inline kyocera_free_shipping_notice/models/sale_order.py:13]: what about 
```python
order.remaining_for_free_shipping = max(0, FREE_SHIP_AMT - order.amount_untaxed)
```

File: kyocera_free_shipping_notice/__manifest__.py [hunk 5]
```diff
@@ -12,7 +12,9 @@
   'author':'Odoo PS',
   'category':'ecommerce',
   'version':'15.0.0.1',
-  'depends':['website_sale'],
+  'depends':[
+    'website_sale'
```
Comment (tdetry) [inline kyocera_free_shipping_notice/__manifest__.py:16]: missing coma (:

**Review by tdetry — APPROVED** [review]
LGTM

**Pull request closed without merging**